src/main/java/com/singlestore/jdbc/plugin/codec/StringCodec.java

```diff
@@ -64,6 +65,11 @@ public boolean canEncode(Object value) {
     return value instanceof String;
   }
 
+  @Override
+  public int getApproximateTextProtocolLength(Object value) throws SQLException {
+    return canEncode(value) ? String.valueOf(value).length() * 3 : -1;
```

Comment:
Why do we multiply by 3 here?

Reply:
is max size possible: 1 char = 2 bytes + escape character = 1 byte encoded (method encode -> writeStringEscaped)